### PR TITLE
Convert error handling into error_chain

### PR DESCRIPTION
**Pull request by farodin91**

### CODE REVIEW

**Comment by RReverser** [comment]
@farodin91 Thanks for your contributions, really appreciated. One small nit: could you please keep changes separated, currently you seem to mix various features and stylistic changes in a single PR which makes it harder and slows down my ability to review them.

Now that this PR is separated from rustfmt changes, it's already much more readable, but it seems to still contain other small changes here and there - some just about preferred codestyle, some slightly change functionality - without reasoning on why they were made.

Thanks!

**Comment by farodin91** [comment]
@RReverser I removed some all unused parse errors from the error_chain. The PR #20 based on this branch, else i had to rebuild all error type conversion by hand. 
Any example for some just about preferred codestyle and some slightly change functionality?

File: src/var.rs [hunk 1]
```diff
@@ -24,17 +24,17 @@ impl<'de, 'a, R: 'a + Read> de::EnumAccess<'de> for EnumAccess<'a, R> {
...
 
             &XmlEvent::Characters(ref name) |
             &XmlEvent::StartElement { name: OwnedName { local_name: ref name, .. }, .. } => {
                 seed.deserialize(name.as_str().into_deserializer())
             }
-        )?;
+        );
         self.de.set_map_value();
-        Ok((name, VariantAccess::new(self.de)))
+        Ok((name?, VariantAccess::new(self.de)))
```
Comment (RReverser) [inline src/var.rs]: @farodin91 For example, here - what is the reason bailout is now done after `set_map_value` and not before it as before?

**Comment by RReverser** [comment]
Left one comment about functionality bit, and as for stylistic changes I think I rather referred to #20, but commented on the wrong PR (looked at both at approximately same time).

File: src/var.rs (hunk 1 above)
Comment (farodin91) [inline src/var.rs]: Resolved. But not clean, because rust cannot infer the type.

File: src/var.rs [hunk 2]
```diff
@@ -24,15 +24,16 @@ impl<'de, 'a, R: 'a + Read> de::EnumAccess<'de> for EnumAccess<'a, R> {
     fn variant_seed<V: DeserializeSeed<'de>>(
         self,
         seed: V,
-    ) -> Result<(V::Value, VariantAccess<'a, R>), Error> {
-        let name = expect!(
+    ) -> Result<(V::Value, VariantAccess<'a, R>)> {
+        let name: Result<V::Value> = expect!(
```
Comment (RReverser) [inline src/var.rs]: You might want to just change this to `let name: V::Value` and then question mark in line should work (or maybe I'm wrong).

Comment (farodin91) [inline src/var.rs]: I found a solution. I updated expect macro

**Comment by farodin91** [comment]
@RReverser Do I have to anything?

**Comment by RReverser** [comment]
Not really, but it's a breaking change, so would rather combine with other major changes and release together.

**Comment by oli-obk** [comment]
@RReverser can we just get it into master? We can stack a few things in master before publishing a breaking change.

@farodin91 needs a rebase/merge

**Comment by RReverser** [comment]
@oli-obk Probably. As, apparently, you've noticed, I haven't had much time 😢  to continue work on this after initial suitable implementation, but generally I don't mind.

**Comment by farodin91** [comment]
@RReverser Rebased

**Comment by oli-obk** [comment]
Thanks!

**Pull request merged**